### PR TITLE
doc-hints-for-docker: gitconfig for private repos. Fixes #4539

### LINKED ISSUE #4539: Private github repos
It might be helpful to inform people in the guide some hints on setting up private repos such as github using the docker container.

```bash
git config --global url."https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com".insteadOf /"https://github.com"
ls -l ~/.gitconfig
```

```bash
docker run -e -v $(pwd):/project \
	   -v ~/.gitconfig:/root/.gitconfig \
            ort:latest --info analyze -f JSON -i /project -o /project/ort/analyzer
 ```

### PR DESCRIPTION
**Pull request by martencassel**
An alternative for .netrc is gitconfig to access private repos.

Signed-off-by: martencassel <marten.cassel@gmail.com>

### CODE REVIEW

**Comment by sschuberth** [comment]
@martencassel, do you plan to work on addressing the review comments?

File: docs/hints-for-use-with-docker.md [hunk 1]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
    ```
 
    **Important:** Ensure that the `.netrc` file has been created at `netrc_folder_path` before running the command, otherwise Docker will create a folder `.netrc` inside the container, instead of mounting a single file.
+
+It is also possible to use ```git config``` to setup access to private repositories, such as github repos.
```
[the remark this answers is not in the document]
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Thanks for the feedback. Changed it as suggested.

File: docs/hints-for-use-with-docker.md [hunk 2]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
...
+
+Configure your gitconfig with your github credentials
+
+    ```shell
+    git config --global url."https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com".insteadOf /"https://github.com"
+    ls -l ~/.gitconfig
+    ```
```
[the remark this answers is not in the document]
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Thanx, changed it as suggested.

File: docs/hints-for-use-with-docker.md [hunk 3]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
...
+When using the container, the .gitconfig can be mounted into the container:
```
[the remark this answers is not in the document]
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Thanx, changed it as you suggested.

File: docs/hints-for-use-with-docker.md [hunk 4]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
...
+
+    ```shell
+    docker run -e -v $(pwd):/project -v ~/.gitconfig:/root/.gitconfig ort --info analyze -f JSON -i /project -o /project/ort/analyzer
```
[the remark this answers is not in the document]
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Thanx, changed it as suggested.

**Comment by martencassel** [comment]
Yes. Updated the PR.

File: docs/hints-for-use-with-docker.md [hunk 5]
```diff
@@ -57,3 +57,24 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
...
+
+    ```shell
+    git config --global url."https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com".insteadOf /"https://github.com"
+    ```
+
+    ```shell
+    cat ~/.gitconfig
+    [url "https://<username>:<password>@github.com"]
+	         insteadOf = /https://github.com    
+    ```
```
Comment (MarcelBochtler) [inline docs/hints-for-use-with-docker.md:73]: I think we should not document pure Git behaviour in this documentation.
A user does not need to run the `git config` command to create the .gitconfig file, they could also create it manually.

````suggestion
1. Configure the `.gitconfig` with the repository credentials. E.g. Github:
    ```
    [url "https://<username>:<password>@github.com"]
             insteadOf = https://github.com    
    ```
````

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]

File: docs/hints-for-use-with-docker.md (hunk 5 above)
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Thanx, implemented changes as suggested.

File: docs/hints-for-use-with-docker.md [hunk 6]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
...
 
    **Important:** Ensure that the `.netrc` file has been created at `netrc_folder_path` before running the command, otherwise Docker will create a folder `.netrc` inside the container, instead of mounting a single file.
+
+It is also possible to use `git config` to setup access to private repositories.
+
+1. Configure your gitconfig with your github credentials
+
+    ```shell
+    [url "https://<username>:<password>@github.com"]
+	         insteadOf = /https://github.com
```
Comment (MarcelBochtler) [inline docs/hints-for-use-with-docker.md:67]: I think the leading `/` in the URL is incorrect.

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]

File: docs/hints-for-use-with-docker.md (hunk 6 above)
Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: Removed it.

**Review by MarcelBochtler — DISMISSED** [review]
LGTM @oss-review-toolkit/core-devs please have a look.

File: docs/hints-for-use-with-docker.md [hunk 7]
```diff
@@ -57,3 +57,19 @@ To authenticate with a private Git repository, ORT uses the (semi)standardized `
    ```
 
    **Important:** Ensure that the `.netrc` file has been created at `netrc_folder_path` before running the command, otherwise Docker will create a folder `.netrc` inside the container, instead of mounting a single file.
+
+It is also possible to use `git config` to setup access to private repositories.
+
+1. Configure your gitconfig with your github credentials
```
Comment (sschuberth) [inline docs/hints-for-use-with-docker.md:63]: Instead of "gitconfig", this should say "`.gitconfig`" as below. Also "github" should be spelled "GitHub".

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: docs/hints-for-use-with-docker.md (hunk 7 above)
Comment (sschuberth) [reply docs/hints-for-use-with-docker.md]: Sorry, my quoting was probably not clear:

    `.github`

Should use backticks in the Markdown file.

Comment (martencassel) [reply docs/hints-for-use-with-docker.md]: "gitconfig" is not written `.gitconfig`
"github" is now spelled "GitHub".

**Comment by sschuberth** [comment]
Merging this without waiting for another CI run to complete.

**Pull request merged**